test/unit/logic/block.js

```diff
@@ -317,40 +326,65 @@ describe('block', () => {
 		done();
 	});
 
-	describe('with valid block and data', () => {
-		beforeEach(done => {
-			data = _.cloneDeep(validDataForBlock);
-			transactions = _.values(transactionsByTypes);
-			done();
+	beforeEach(done => {
+		data = _.cloneDeep(validDataForBlock);
+		transactions = _.values(transactionsByTypes);
+		done();
+	});
+
+	describe('create', () => {
+		var blockNormalizeStub;
+
+		before(() => {
+			blockNormalizeStub = sinonSandbox
+				.stub(block, 'objectNormalize')
+				.returnsArg(0);
+
+			transactionStub.getBytes.returns(Buffer.from('dummy transaction bytes'));
+			return transactionStub.objectNormalize.returnsArg(0);
 		});
 
-		describe('create', () => {
-			var blockNormalizeStub;
+		after(() => {
+			blockNormalizeStub.reset();
+			transactionStub.getBytes.reset();
+			return transactionStub.objectNormalize.reset();
+		});
 
-			before(() => {
-				blockNormalizeStub = sinonSandbox
-					.stub(block, 'objectNormalize')
-					.returnsArg(0);
+		describe('when one of all transaction types are present', () => {
+			var generatedBlock;
+			var transactionsOrder;
+			var correctOrder = [0, 1, 2, 3, 5, 6, 7, 4];
 
-				transactionStub.getBytes.returns(
-					Buffer.from('dummy transaction bytes')
-				);
-				return transactionStub.objectNormalize.returnsArg(0);
+			beforeEach(done => {
+				data.transactions = transactions;
+				generatedBlock = block.create(data);
+				transactionsOrder = generatedBlock.transactions.map(trs => {
+					return trs.type;
+				});
+				done();
 			});
 
-			after(() => {
-				blockNormalizeStub.reset();
-				transactionStub.getBytes.reset();
-				return transactionStub.objectNormalize.reset();
+			it('should sort transactions in the correct order', () => {
+				expect(generatedBlock.transactions.length).to.equal(
+					data.transactions.length
+				);
+				return expect(transactionsOrder).to.eql(correctOrder);
 			});
+		});
+
+		describe('when there are multiple multisignature transactions', () => {
+			var correctOrderOfTransactions = [0, 1, 2, 3, 5, 6, 7, 4, 4, 4, 4, 4, 4];
 
-			describe('when one of all transaction types are present', () => {
+			describe('in the beginning', () => {
+				var multipleMultisigTx;
 				var generatedBlock;
 				var transactionsOrder;
-				var correctOrder = [0, 1, 2, 3, 5, 6, 7, 4];
 
 				beforeEach(done => {
-					data.transactions = transactions;
+					multipleMultisigTx = Array(...Array(5)).map(() => {
```

Comment:
add comments for things like that, for example - `// Create 6 multisignature transactions`